### PR TITLE
CMSIS-NN: Add dilation support for conv

### PR DESCRIPTION
**Pull request by mansnils**
Adds dilation support for the conv kernel.
Adds corresponding unit tests for dilation as well.
Also fixes generate script bug that it was always regenerating
data.

### CODE REVIEW

**Comment by grasci-arm** [comment]
Can one of the admins verify this patch?

File: ARM.CMSIS.pdsc [hunk 1]
```diff
@@ -14,6 +14,7 @@
       CMSIS-NN: 3.1.0 (see revision history for details)
        - Support for int16 convolution and fully connected for reference implementation
        - Support for DSP extension optimization for int16 convolution and fully connected
+       - Support for dilation for int18 convolution
```
Comment (felix-johnny) [inline ARM.CMSIS.pdsc:17]: int8.

File: CMSIS/DoxyGen/NN/src/history.txt [hunk 2]
```diff
@@ -19,6 +19,7 @@
       <li> Added fully_connected int16 reference implementation </li>
       <li> Added fully_connected int16 DSP implementation </li>
       <li> Added unit tests for int16 fully_connected kernel </li>
+      <li> Added dilation support for int8 conv kernel </li>
```
Comment (felix-johnny) [inline CMSIS/DoxyGen/NN/src/history.txt:22]: Please update the table in https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN#tfl-micro-compliant-apis as well.

File: ARM.CMSIS.pdsc (hunk 1 above)
Comment (mansnils) [reply ARM.CMSIS.pdsc]: done

File: CMSIS/DoxyGen/NN/src/history.txt (hunk 2 above)
Comment (mansnils) [reply CMSIS/DoxyGen/NN/src/history.txt:22]: done

**Pull request merged**